### PR TITLE
[WIP]feat: use new project api

### PR DESCRIPTION
**Pull request by andre-code**
This PR includes only adapting the current "create new project" form to use react-hook-form and fetch data using redux-toolkit. 

Features:
- [x] Set name, namespace, description, avatar, visibility, user template, and template.
- [x] Fetch/refetch namespaces.
- [x] Calculate slug by name and namespace.
- [x] Set visibility by namespace.
- [x] Fetch custom and RenkuLab templates.
- [x] Set template variables (e.g., Pinned image template).
- [x] Restore default template variable.

Form validations:
- [x] Invalid project name.
- [x] Duplicated project name.

Pending to include in this PR:
- [ ] Show a message fetching data next to the submit button.
- [ ] Show errors fetching templates.
- [ ] Validate required fields (title, namespace, template) and display them at the end of the form.

/deploy #persist

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -187,15 +187,9 @@ function CentralContentContainer(props) {
           <Route
             exact
             path={Url.get(Url.pages.project.new)}
-            render={(p) => (
+            render={() => (
```
Comment (leafty) [inline client/src/App.js:190]: In that case, use:
```tsx
<Route
  exact
  path={Url.get(Url.pages.project.new)}
>
  <ContainerWrap>
    <NewProject />
  </ContainerWrap>
</Route>
```

File: client/src/features/project/components/form/NamespaceInput.tsx
Comment (leafty) [inline client/src/features/project/components/form/NamespaceInput.tsx:1]: Why separate this as components? It looks tedious to get the types for the props and this is not reusable as these components are only compatible with the "New Project" form.

File: client/src/features/project/editNew/NewProject.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import { SubmitHandler } from "react-hook-form";
+import { NewProjectFormFields } from "../projectKg.types";
+import { CreateProjectParams } from "../Project";
+import FormSchema from "../../../components/formschema/FormSchema";
+import { useLocation } from "react-router-dom";
+import { redirectAfterSubmit } from "./NewProject.utils";
+import { useHistory } from "react-router";
+
+export function NewProject() {
+  const [submitting, setSubmitting] = React.useState(false);
```
Comment (leafty) [inline client/src/features/project/editNew/NewProject.tsx:36]: Why is this not coming from the mutation hook?

File: client/src/features/project/editNew/NewProject.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const dispatch = useAppDispatch();
+  const [createProjectMutation] = useCreateProjectMutation();
+  const formState = useAppSelector(({ projectForm }) => projectForm);
+  const user = useLegacySelector((state) => state.stateModel.user);
+  const location = useLocation();
+  const history = useHistory();
+  const onSubmit: SubmitHandler<NewProjectFormFields> = React.useCallback(
+    async (data: NewProjectFormFields) => {
```
Comment (leafty) [inline client/src/features/project/editNew/NewProject.tsx:44]: Code smell: async hook. Async hooks are hard to get right, prefer to use the state from the mutation hook.

File: client/src/features/project/editNew/NewProject.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            visibility: submitData.visibility,
+          },
+        };
+
+        const response = await createProjectMutation(newProjectBody);
+        // TODO: Use progress component and show success or error message
+        setSubmitting(false);
+        if ("data" in response && response.data.message === "Project created") {
+          const projectPathWithNamespace = response.data.slug || "";
+          redirectAfterSubmit(history, projectPathWithNamespace, undefined);
```
Comment (leafty) [inline client/src/features/project/editNew/NewProject.tsx:69]: Why is this a util? This should be a rendered `<Redirect>` or found inside a `useEffect()` hook (consuming `isSuccess` and the data from the mutation hook`).

**Review by leafty — COMMENTED** [review]
Some early remarks:

**Comment by andre-code** [comment]
No longer needed

**Pull request closed without merging**